### PR TITLE
Add Build Blog Action

### PR DESCRIPTION
**Pull request by aeilot**

### CODE REVIEW

File: .github/workflows/blog-builder.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: "Hexo Blog Builder"
+
+on: [push]
+
+jobs:
+  format:
+    name: "Run Clang Format"
+    runs-on: ubuntu-latest
```
Comment (aeilot) [inline .github/workflows/blog-builder.yml:8]: Can also run on Windows and macOS

File: .github/workflows/blog-builder.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: npm install
+      - name: Hexo Clean
+        run: hexo clean
+      - name: Hexo Build
+        run: hexo g
+      - name: Deploy
+        uses: y4code/hexo-deploy-action@master
+        env:
+          PERSONAL_TOKEN: ${{ secrets.BLOG_BUILDER_ACTION_SECRET }}
```
Comment (aeilot) [inline .github/workflows/blog-builder.yml:34]: Your token

File: .github/workflows/blog-builder.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          PUBLISH_REPOSITORY: MrWillCom/MrWillCom.github.io
```
Comment (aeilot) [inline .github/workflows/blog-builder.yml:35]: Publish Repo

File: .github/workflows/blog-builder.yml (hunk 1 above)
Comment (MrWillCom) [reply .github/workflows/blog-builder.yml:8]: Why format it?

File: .github/workflows/blog-builder.yml (hunk 2 above)
Comment (MrWillCom) [reply .github/workflows/blog-builder.yml:34]: Does y4code/hexo-deploy-action safe?

File: .github/workflows/blog-builder.yml (hunk 3 above)
Comment (MrWillCom) [reply .github/workflows/blog-builder.yml:35]: It doesn't works.

**Pull request merged**